dags/openshift_nightlies/scripts/install/hypershift.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+set -ex
+
+while getopts v:a:j:o: flag
+do
+    case "${flag}" in
+        v) version=${OPTARG};;
+        j) json_file=${OPTARG};;
+        o) operation=${OPTARG};;
+        *) echo "ERROR: invalid parameter ${flag}" ;;
+    esac
+done
+
+_get_cluster_id(){
+    echo $(rosa list clusters -o json | jq -r '.[] | select(.name == '\"$1\"') | .id')
+}
+
+_download_kubeconfig(){
+    ocm get /api/clusters_mgmt/v1/clusters/$1/credentials | jq -r .kubeconfig > $2
+}
+
+_get_cluster_status(){
+    echo $(rosa list clusters -o json | jq -r '.[] | select(.name == '\"$1\"') | .status.state')
+}
+
+_get_base_domain(){
+    ocm get /api/clusters_mgmt/v1/clusters/$1/ | jq -r '.dns.base_domain'
+}
+
+setup(){
+    mkdir /home/airflow/workspace
+    cd /home/airflow/workspace
+    export PATH=$PATH:/usr/bin:/usr/local/go/bin:/opt/airflow/hypershift/bin
+    export HOME=/home/airflow
+    export AWS_REGION=us-west-2
+    export AWS_ACCESS_KEY_ID=$(cat ${json_file} | jq -r .aws_access_key_id)
+    export AWS_SECRET_ACCESS_KEY=$(cat ${json_file} | jq -r .aws_secret_access_key)
+    export ROSA_ENVIRONMENT=$(cat ${json_file} | jq -r .rosa_environment)
+    export ROSA_TOKEN=$(cat ${json_file} | jq -r .rosa_token_${ROSA_ENVIRONMENT})
+    export MGMT_CLUSTER_NAME=$(cat ${json_file} | jq -r .openshift_cluster_name)
+    export HOSTED_CLUSTER_NAME=$MGMT_CLUSTER_NAME-$HOSTED_NAME
+    export HOSTED_KUBECONFIG_NAME=$(echo $KUBECONFIG_NAME | awk -F-kubeconfig '{print$1}')-$HOSTED_NAME-kubeconfig
+    export HOSTED_KUBEADMIN_NAME=$(echo $KUBEADMIN_NAME | awk -F-kubeadmin '{print$1}')-$HOSTED_NAME-kubeadmin    
+    export KUBECONFIG=/home/airflow/auth/config
+    export ROSA_CLI_VERSION=$(cat ${json_file} | jq -r .rosa_cli_version)
+    if [[ ${ROSA_CLI_VERSION} == "master" ]]; then
+        git clone https://github.com/openshift/rosa
+	pushd rosa
+	make
+	sudo mv rosa /usr/local/bin/
+	popd
+    fi
+    echo [default] >> aws_credentials
+    echo aws_access_key_id=$AWS_ACCESS_KEY_ID >> aws_credentials
+    echo aws_secret_access_key=$AWS_SECRET_ACCESS_KEY >> aws_credentials
+    rosa login --env=${ROSA_ENVIRONMENT}
+    ocm login --url=https://api.stage.openshift.com --token="${ROSA_TOKEN}"
+    rosa whoami
+    rosa verify quota
+    rosa verify permissions
+    echo "MANAGEMENT CLUSTER VERSION:"
+    ocm list cluster $MGMT_CLUSTER_NAME
+    echo "MANAGEMENT CLUSTER NODES:"
+    kubectl get nodes
+}
+
+install(){
+    echo "Install Hypershift Operator"
+    # aws s3api delete-bucket --bucket $MGMT_CLUSTER_NAME-aws-rhperfscale-org --region $AWS_REGION || true
+    aws s3api create-bucket --acl public-read --bucket $MGMT_CLUSTER_NAME-aws-rhperfscale-org --create-bucket-configuration LocationConstraint=$AWS_REGION --region $AWS_REGION || true
+    sleep 10 # wait a few seconds 
```

Comment:
what are we specifically waiting for? is it something we can query for a ready state?

Reply:
yes just a random sleep to make sure its available, I will see if I can add a valid check.